scserver.js

```diff
@@ -533,6 +533,9 @@ SCServer.prototype._handleSocketConnection = function (wsSocket, upgradeReq) {
         respond(err, clientSocketErrorStatus);
         scSocket.disconnect(statusCode);
         return;
+      } else if (!signedAuthToken && scSocket.signedAuthToken) {
```

Comment:
- The `else` is not needed because the previous line has a `return`.
- What about the case where `signedAuthToken` was set before but `scSocket.setAuthToken(...)` was called inside `MIDDLEWARE_HANDSHAKE_SC`; it should replace it.

Reply:
thanks for the review updated